src/server/response.cpp

```diff
@@ -93,15 +93,15 @@ std::unique_ptr<Response> Response::build_304(const InternalServer& server, cons
   return response;
 }
 
-std::unique_ptr<Response> Response::build_404(const InternalServer& server, const RequestContext& request, const std::string& bookName)
+std::unique_ptr<Response> Response::build_404(const InternalServer& server, const RequestContext& request, const std::string& bookName, const std::string& pattern)
 {
   MustacheData results;
   results.set("url", request.get_full_url());
-
+  results.set("bookName", bookName);
   auto response = ContentResponse::build(server, RESOURCE::templates::_404_html, results, "text/html");
   response->set_code(MHD_HTTP_NOT_FOUND);
   response->set_taskbar(bookName, "");
-
+  response->set_search_link(bookName, pattern);
```

Comment:
@veloman-yunkan already told you to not modify the `ContentResponse` object but I continue :)

You should simply modify the `_404_html` template to add a details "token/variable" and add the "details" variable to `results`.